backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/payment_allocation/process/PaymentsViewAllocateCommand.java

```diff
@@ -210,9 +209,17 @@ static PayableDocument toPayableDocument(
 						.build()
 						.negateIf(soTrx.isPurchase()))
 				.invoiceProcessingFeeCalculation(invoiceProcessingFeeCalculation)
+				.date(row.getDateInvoiced())
+				.clientAndOrgId(row.getClientAndOrgId())
+				.currencyConversionTypeId(row.getCurrencyConversionTypeId())
 				.build();
 	}
 
+	private static boolean hasServiceFeesToPay(final @NonNull InvoiceRow row, final CurrencyCode currencyCode)
+	{
+		return row.getServiceFeeAmt() != null && !Amount.zero(currencyCode).equals(row.getServiceFeeAmt());
```

Comment:
why not using Amount.isZero instead of that weird "equals" comparation....

Reply:
done